Fail to build docker

### Discussed in https://github.com/danny-avila/LibreChat/discussions/1329

<div type='discussions-op-text'>

<sup>Originally posted by **Originalimoc** December 11, 2023</sup>
### Contact Details

_No response_

### What happened?

docker-compose build: WARN BADENGINE
docker-compose up: Error: Cannot find module 'langchain/schema'

### Steps to Reproduce

git pull, docker-compose build.
vite require "^18.0.0 | >=20.0.0" npm but installed 19.9

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Screenshots

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct</div>

> WARN BADENGINE

Nothing to worry about here. Vite is pushing this warning because we are on node v19 and it has reach its End of life, as it will no longer be maintained; therefore, vite will not raise concern over issues using this engine. This is very recent, and we will move to node:20-alpine instead of 19 soon once I can debug other unrelated issues related to this upgrade with the default compose file.

> Error: Cannot find module 'langchain/schema'

This indicates to me that the image is out of sync with the other discussion's issue `Error: Cannot find module '/app/api/server/index.js'` (https://github.com/danny-avila/LibreChat/discussions/776)

 I was able to reproduce the issue on a computer on an old commit and it's an issue with the compose file, should be fixed with this: https://github.com/danny-avila/LibreChat/pull/1330

--- a/config/update.js
+++ b/config/update.js
@@ -97,7 +97,7 @@ async function validateDockerRunning() {
     console.purple('Building new LibreChat image...');
     const buildCommand = `${sudo}docker-compose ${
       singleCompose ? '-f ./docs/dev/single-compose.yml ' : ''
-    }build`;
+    }build --no-cache`;
     console.orange(buildCommand);
     execSync(buildCommand, { stdio: 'inherit' });
   } else {

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -38,7 +38,7 @@ services:
       - MEILI_HOST=http://meilisearch:7700
       - MEILI_HTTP_ADDR=meilisearch:7700
     volumes:
-      - /app/client/node_modules
+      - /app/client/node_modules # node_modules mapping necessary for module persistence
       - /app/api/node_modules
       - ./api:/app/api
       - ./.env:/app/.env

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -38,6 +38,8 @@ services:
       - MEILI_HOST=http://meilisearch:7700
       - MEILI_HTTP_ADDR=meilisearch:7700
     volumes:
+      - /app/client/node_modules
+      - /app/api/node_modules
       - ./api:/app/api
       - ./.env:/app/.env
       - ./.env.development:/app/.env.development

